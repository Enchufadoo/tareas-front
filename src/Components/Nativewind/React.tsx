import { styled as nativewindStyled } from 'nativewind'
import * as ReactNative from 'react-native'
import * as VectorIcons from '@expo/vector-icons'
import { styled } from 'dripsy'

export const Text = styled(nativewindStyled(ReactNative.Text))({
  color: '$text'
})

export const View = styled(nativewindStyled(ReactNative.View))()

export const Pressable = styled(nativewindStyled(ReactNative.Pressable))()
export const TextInput = styled(nativewindStyled(ReactNative.TextInput))()
export const TouchableOpacity = styled(
  nativewindStyled(ReactNative.TouchableOpacity)
)()
export const Switch = styled(nativewindStyled(ReactNative.Switch))()
export const SimpleLineIcons = styled(
  nativewindStyled(VectorIcons.SimpleLineIcons)
)()
export const MaterialIcons = styled(
  nativewindStyled(VectorIcons.MaterialIcons)
)()

export const Entypo = styled(nativewindStyled(VectorIcons.Entypo))()

export const Octicons = styled(nativewindStyled(VectorIcons.Octicons))()
