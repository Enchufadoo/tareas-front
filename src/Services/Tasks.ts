import { api } from './Api'
import moment from 'moment'

export type Task = {
  id: number
  title: string
  description: string
  created_at: string
  created_at_date: string
  hours_since_progress: number
  progress_today: boolean
  last_progress?: {
    created_at: string
    id: number
  }
}

export type TaskCount = { finished: number; unfinished: number }

export enum TaskProgressState {
  open,
  finished,
  all
}

interface TaskWithUser extends Task {
  user: {
    username: string
    email: string
    avatar: string
  }
}

export type GetTasksResponse = {
  data: {
    tasks: Task[]
    task_count: TaskCount
  }
}

type GetTaskResponse = {
  data: {
    task: TaskWithUser
  }
}

type getCreateTaskDataResponse = {
  data: {
    friends: string[]
  }
}
type setAddTaskProgressResponse = {
  data: {}
}
type setRemoveTaskProgressResponse = {
  data: {}
}

type setDeleteTaskResponse = {
  data: {}
}

type CreateTaskData = {
  title: string
  description: string
}

export enum FinishTask {
  unfinished,
  finished
}

type FinishTaskData = {
  taskId: number
  finishTask: FinishTask
}

type setCreateTaskResponse = {}

export const tasksApi = api.injectEndpoints({
  overrideExisting: true,
  endpoints: (build) => ({
    getTasks: build.query<GetTasksResponse, number>({
      providesTags: ['TasksList'],
      query: (finished: number) => {
        return {
          url: `/task`,
          method: 'GET',
          params: {
            finished: finished
          }
        }
      }
    }),
    getTask: build.query<GetTaskResponse, number>({
      query: (id: number) => {
        return {
          url: `/task/${id}`,
          method: 'GET'
        }
      },
      transformResponse: (response: GetTaskResponse) => {
        response.data.task.created_at_date = moment().format('YYYY-MM-DD')
        return response
      },
      providesTags: ['TaskView']
    }),
    setCreateTask: build.mutation<setCreateTaskResponse, CreateTaskData>({
      query: (taskData) => {
        return {
          url: `/task`,
          method: 'POST',
          data: taskData
        }
      },
      invalidatesTags: ['TasksList']
    }),
    setAddTaskProgress: build.mutation<setAddTaskProgressResponse, number>({
      query: (taskId) => {
        return {
          url: `/task/progress/${taskId}`,
          method: 'POST',
          data: {}
        }
      },
      invalidatesTags: ['TaskView', 'TasksList']
    }),
    setRemoveTaskProgress: build.mutation<
      setRemoveTaskProgressResponse,
      number
    >({
      query: (taskId) => {
        return {
          url: `/task/progress/${taskId}`,
          method: 'DELETE',
          data: {}
        }
      },
      invalidatesTags: ['TaskView', 'TasksList']
    }),
    setDeleteTask: build.mutation<setDeleteTaskResponse, number>({
      query: (taskId) => {
        return {
          url: `/task/${taskId}`,
          method: 'DELETE',
          data: {}
        }
      },
      invalidatesTags: ['TasksList']
    }),
    setFinishTask: build.mutation<setDeleteTaskResponse, FinishTaskData>({
      query: (finishTaskData) => {
        return {
          url: `/task/finish/${finishTaskData.taskId}/${finishTaskData.finishTask}`,
          method: 'PUT',
          data: {}
        }
      },
      invalidatesTags: ['TasksList']
    }),
    getCreteTaskData: build.query<getCreateTaskDataResponse, void>({
      query: () => {
        return {
          url: `/task/store/data`,
          method: 'GET'
        }
      }
    }),
    getWeeksTasks: build.query<
      {
        data: {
          progress: {
            [key: number]: boolean[]
          }
          tasks: {
            id: number
            title: string
          }[]
        }
      },
      void
    >({
      query: () => {
        return {
          url: `/task/week`,
          method: 'GET'
        }
      },
      providesTags: []
    })
  })
})

export const {
  useGetTasksQuery,
  useGetTaskQuery,
  useGetCreteTaskDataQuery,
  useSetCreateTaskMutation,
  useSetAddTaskProgressMutation,
  useSetRemoveTaskProgressMutation,
  useSetFinishTaskMutation,
  useLazyGetWeeksTasksQuery
} = tasksApi
