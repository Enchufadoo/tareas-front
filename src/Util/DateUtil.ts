import moment, { Moment } from 'moment'

export const ellapsedTime = (datetime: string) => {
  let end = moment(datetime)
  let start = moment()

  let days = Math.floor(moment.duration(start.diff(end)).asDays())

  if (!days) return 'today'

  if (days === 1) return 'yesterday'

  return days + ' days ago'
}

export const hours = (dateObj: moment.Moment): string => {
  return dateObj.format('HH:mm')
}
