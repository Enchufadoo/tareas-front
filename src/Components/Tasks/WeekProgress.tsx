import React from 'react'

import { Entypo, Text, View } from '@/Components/Nativewind/React'
import { StyleProp } from 'react-native/Libraries/StyleSheet/StyleSheet'
import { ViewStyle } from 'react-native/Libraries/StyleSheet/StyleSheetTypes'
import { useSx } from 'dripsy'

const BORDER_WIDTH = 0.3

type Props = {
  progress: boolean[]
  isLastElement: boolean
}

const WeekProgress = (props: Props) => {
  const sx = useSx()

  let rowStyle: StyleProp<ViewStyle> = {
    flexDirection: 'row',
    borderTopWidth: BORDER_WIDTH
  }

  if (props.isLastElement) {
    rowStyle.borderBottomWidth = BORDER_WIDTH
  }

  return (
    <View style={rowStyle}>
      {props.progress.map((day, index) => {
        let style: StyleProp<ViewStyle> = {
          borderRightWidth: BORDER_WIDTH,
          borderColor: 'black'
        }

        if (index === 0) {
          style.borderLeftWidth = BORDER_WIDTH
        }
        return (
          <View
            key={index}
            className={'justify-center items-center p-2 flex-1'}
            style={style}
          >
            {day ? (
              <Entypo
                name={'cross'}
                size={24}
                color={sx({ color: '$success' })['color']}
              />
            ) : (
              <Text>{''}</Text>
            )}
          </View>
        )
      })}
    </View>
  )
}

export default WeekProgress
