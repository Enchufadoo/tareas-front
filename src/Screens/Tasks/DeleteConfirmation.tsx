import React from 'react'
import { View } from 'react-native'

type DialogComponentProps = {
  visible: boolean
  toggleVisible: () => void
  deleteAction: () => void
}

const DeleteConfirmation: React.FunctionComponent<DialogComponentProps> = (
  props
) => {
  return (
    <View>
      {/*<Dialog isVisible={props.visible} onBackdropPress={props.toggleVisible}>
        <Dialog.Title title="Delete Task Confirmation" />
        <Text>Are you sure you want to delete this task</Text>
        <Dialog.Actions>
          <Dialog.Button title="Delete" onPress={props.deleteAction} />
          <Dialog.Button title="Cancel" onPress={() => props.toggleVisible()} />
        </Dialog.Actions>
      </Dialog>*/}
    </View>
  )
}

export default DeleteConfirmation
