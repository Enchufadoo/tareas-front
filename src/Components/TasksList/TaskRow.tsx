import React, { useRef, useState } from 'react'
import { Animated, StyleSheet } from 'react-native'

import { capitalize } from '@/Util/StringUtil'

import Ionicons from '@expo/vector-icons/Ionicons'
import { RectButton } from 'react-native-gesture-handler'
import Swipeable from 'react-native-gesture-handler/Swipeable'
import { Octicons, Pressable, Text, View } from '../Nativewind/React'
import {
  Task,
  useSetAddTaskProgressMutation,
  useSetRemoveTaskProgressMutation
} from '@/Services/Tasks'
import { ellapsedTime, hours } from '@/Util/DateUtil'
import moment from 'moment'
import { jc } from '@/Util/NativewindUtil'

const AnimatedView = Animated.createAnimatedComponent(View)

type TaskRowProps = {
  onNavigateToTask: () => void
  index: number
  item: Task
}

export default function TaskRow(props: TaskRowProps) {
  const [isOpen, setIsOpen] = useState(false)

  const [setAddTaskProgress, setAddTaskProgressResult] =
    useSetAddTaskProgressMutation()

  const [setRemoveTaskProgress, setRemoveTaskProgressResult] =
    useSetRemoveTaskProgressMutation()

  let swipeableRow = useRef(null)

  const updateRef = (ref: Swipeable) => {
    swipeableRow.current = ref
  }
  const close = () => {
    swipeableRow.current.close()
  }

  const renderLeftActions = (
    _progress: Animated.AnimatedInterpolation<number>,
    dragX: Animated.AnimatedInterpolation<number>
  ) => {
    const scale = dragX.interpolate({
      inputRange: [0, 20],
      outputRange: [0, 1],
      extrapolate: 'clamp'
    })

    let style = !props.item.progress_today
      ? styles.leftAddAction
      : styles.leftCloseAction

    let icon: 'close' | 'checkmark-done' = !props.item.progress_today
      ? 'checkmark-done'
      : 'close'

    return (
      <RectButton style={style} onPress={close}>
        <AnimatedView style={[styles.actionIcon, { transform: [{ scale }] }]}>
          <Ionicons name={icon} size={32} color="white" />
        </AnimatedView>
      </RectButton>
    )
  }

  const renderRowContent = (c: Task) => {
    let mainClasses = [
      'shadow-sm',
      'lg:rounded-md',
      'flex-column',
      'rounded-sm',
      '-ml-2'
    ]

    let progressDate = c.last_progress?.created_at
      ? moment(c.last_progress.created_at)
      : null

    return (
      <Pressable
        onPress={() => {
          props.onNavigateToTask()
        }}
        className={jc(mainClasses)}
        style={{ elevation: 1, padding: 0, paddingLeft: 0 }}
        sx={{ backgroundColor: '$background' }}
        key={c.id.toString()}
      >
        <View className="ml-3 mr-3">
          <View className="flex-row ">
            <View className="flex-1">
              <Text
                className="text-xl font-medium truncate"
                sx={{
                  color: '$text'
                }}
              >
                {c.title}
              </Text>
              <Text className="text-base truncate" sx={{ color: '$lightText' }}>
                {c.description}
              </Text>
            </View>
            <View className="justify-center">
              {props.item.progress_today && (
                <Octicons sx={{ color: '$success' }} name="check" size={24} />
              )}
            </View>
          </View>
        </View>
        <View className={'ml-3 mr-3'}>
          <View
            className="h-2 w-full mb-2  "
            style={{
              borderBottomColor: '#AAA',
              borderBottomWidth: StyleSheet.hairlineWidth
            }}
          ></View>
          <View>
            {progressDate ? (
              <Text className="flex-row space-x-1 text-sm py-1">
                <Text>
                  {capitalize(ellapsedTime(c.last_progress.created_at))}
                </Text>
                <Text>{' ' + hours(progressDate)}</Text>
              </Text>
            ) : (
              <View>
                <Text className="text-sm py-1" sx={{ color: '$lightText' }}>
                  No progress yet
                </Text>
              </View>
            )}
          </View>
        </View>
      </Pressable>
    )
  }

  return (
    <Swipeable
      ref={updateRef}
      friction={3}
      leftThreshold={20}
      enableTrackpadTwoFingerGesture
      renderLeftActions={renderLeftActions}
      overshootLeft={false}
      onSwipeableOpen={close}
      containerStyle={{ marginBottom: 10 }}
      onSwipeableWillOpen={() => {
        setIsOpen(true)
      }}
      onSwipeableClose={() => {
        if (isOpen) {
          if (props.item.progress_today) {
            setRemoveTaskProgress(props.item.id)
          } else {
            setAddTaskProgress(props.item.id)
          }

          setIsOpen(false)
        }
      }}
    >
      {renderRowContent(props.item)}
    </Swipeable>
  )
}

const styles = StyleSheet.create({
  leftAddAction: {
    flex: 1,
    maxWidth: 80,
    width: '100%',
    backgroundColor: '#5c5de5',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row'
  },
  leftCloseAction: {
    flex: 1,
    maxWidth: 80,
    width: '100%',
    backgroundColor: '#e10808',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row'
  },
  actionIcon: {}
})
