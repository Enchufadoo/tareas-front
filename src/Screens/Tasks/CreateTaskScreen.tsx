import { Text, View } from '@/Components/Nativewind/React'
import {
  useGetCreteTaskDataQuery,
  useSetCreateTaskMutation
} from '@/Services/Tasks'
import { setLoadingData } from '@/Store/Features/ApplicationSlice'
import { NativeStackScreenProps } from '@react-navigation/native-stack'
import React, { useEffect } from 'react'
import { ScrollView } from 'react-native'
import { useDispatch } from 'react-redux'
import { Controller, useForm } from 'react-hook-form'
import { isValidForm } from '@/Util/FormUtil'
import InputError from '@/Components/Errors/InputError'

import Button, {
  ButtonSize,
  ButtonTheme
} from '@/Components/Base/Button/Button'
import FormSubmitError from '@/Components/Errors/FormSubmitError'
import { MultilineInput } from '@/Components/Base/Input/MultilineInput'
import { TextInput } from '@/Components/Base/Input/TextInput'
import BaseLayout from '@/Components/Layout/BaseLayout/BaseLayout'
import { LoggedInStackParamList } from '@/Router/LoggedInStackNavigator'

type Props = NativeStackScreenProps<LoggedInStackParamList, 'Create Task'>

const CreateTaskScreen = (props: Props) => {
  const { isFetching, isSuccess, isError, data } = useGetCreteTaskDataQuery()

  const [setCreateTask, setCreateTaskResult] = useSetCreateTaskMutation()

  const dispatch = useDispatch()

  const requiredFields = ['title']

  const {
    getValues,
    control,
    formState: { errors, isDirty, dirtyFields }
  } = useForm({
    mode: 'onChange',
    defaultValues: {
      title: '',
      description: ''
    }
  })

  let isValid = isValidForm(errors, requiredFields, dirtyFields)

  useEffect(() => {
    if (setCreateTaskResult.isSuccess) {
      props.navigation.navigate('Home Screen')
    }
  }, [setCreateTaskResult.isSuccess])

  useEffect(() => {
    dispatch(setLoadingData(isFetching))
  }, [isFetching])

  return (
    <>
      {isSuccess && (
        <ScrollView style={{ flex: 1 }} contentContainerStyle={{ flexGrow: 1 }}>
          <BaseLayout>
            <View className="flex-1">
              <View className="mb-5">
                <Text className="text-2xl font-bold">New Task</Text>
                <Text>Create a new task</Text>
              </View>
              <View className="flex-1 space-y-4">
                <View>
                  <Controller
                    control={control}
                    rules={{
                      required: {
                        value: true,
                        message: 'The title field is required'
                      },
                      minLength: {
                        value: 4,
                        message: 'The title must be at least 4 characters long'
                      },
                      maxLength: {
                        value: 30,
                        message: 'The title must be at most 30 characters'
                      }
                    }}
                    render={({ field: { onChange, onBlur, value } }) => (
                      <TextInput
                        placeholder={'Title'}
                        onBlurForm={onBlur}
                        onChangeText={onChange}
                        value={value}
                        keyboardType="default"
                        isValid={control.getFieldState('title')}
                      />
                    )}
                    name="title"
                  />

                  <InputError error={errors.title} />
                </View>
                <View>
                  <Controller
                    control={control}
                    rules={{
                      required: {
                        value: false,
                        message: 'The description field is required'
                      },
                      minLength: {
                        value: 4,
                        message:
                          'The description must be at least 4 characters long'
                      },
                      maxLength: {
                        value: 3000,
                        message: 'The title must be at most 3000 characters'
                      }
                    }}
                    render={({ field: { onChange, onBlur, value } }) => (
                      <MultilineInput
                        height={300}
                        placeholder={'Description'}
                        onBlurForm={onBlur}
                        onChangeText={onChange}
                        value={value}
                        keyboardType="default"
                        isValid={control.getFieldState('description')}
                      />
                    )}
                    name="description"
                  />

                  <InputError error={errors.title} />
                </View>
                {setCreateTaskResult.isError && (
                  <View className={'mt-6'}>
                    <FormSubmitError
                      header={'Creating a task failed'}
                      subheader={'Please try again later'}
                    />
                  </View>
                )}
              </View>
            </View>
            <View className={'mt-12 mb-12'}>
              <Button
                theme={ButtonTheme.primary}
                size={ButtonSize.medium}
                title={'Create Task'}
                disabled={!isDirty || !isValid}
                outline={false}
                loading={
                  setCreateTaskResult.isLoading || setCreateTaskResult.isSuccess
                }
                onPress={async () => {
                  await setCreateTask(getValues())
                }}
              />
            </View>
          </BaseLayout>
        </ScrollView>
      )}
    </>
  )
}

export default CreateTaskScreen
