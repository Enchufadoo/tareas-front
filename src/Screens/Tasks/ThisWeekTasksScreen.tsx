import React, { useCallback } from 'react'
import BaseLayout from '@/Components/Layout/BaseLayout/BaseLayout'
import { Text, View } from '@/Components/Nativewind/React'
import { useLazyGetWeeksTasksQuery } from '@/Services/Tasks'
import { useLoadingData } from '@/Lib/UseLoadingData'
import { useFocusEffect } from '@react-navigation/native'
import WeekProgress from '@/Components/Tasks/WeekProgress'

const ThisWeekTasksScreen = () => {
  const [getWeeksTasksTrigger, getWeeksTasksResult] =
    useLazyGetWeeksTasksQuery()

  useFocusEffect(
    useCallback(() => {
      getWeeksTasksTrigger()
    }, [])
  )

  useLoadingData([getWeeksTasksResult])

  return (
    <BaseLayout>
      <Text className="text-2xl font-bold">Progress This Week</Text>

      {getWeeksTasksResult.isSuccess &&
        !getWeeksTasksResult.isFetching &&
        (getWeeksTasksResult.data?.data?.tasks?.length ? (
          <View className={'flex-1 my-2'}>
            <View className="flex-row">
              <View style={{ flex: 1 }}></View>
              <View style={{ flex: 3 }}>
                <View className={'flex-row'}>
                  {['S', 'M', 'T', 'W', 'T', 'F', 'S'].map((day, index) => {
                    return (
                      <View className={'flex-1 items-center'} key={index}>
                        <Text>{day}</Text>
                      </View>
                    )
                  })}
                </View>
              </View>
            </View>
            {getWeeksTasksResult.data.data.tasks.map((task, index) => {
              return (
                <View className={'flex-row'} key={task.id}>
                  <View className={'flex-1 justify-center '}>
                    <Text numberOfLines={2}>{task.title}</Text>
                  </View>
                  <View style={{ flex: 3 }}>
                    <WeekProgress
                      progress={getWeeksTasksResult.data.data.progress[task.id]}
                      isLastElement={
                        index === getWeeksTasksResult.data.data.tasks.length - 1
                      }
                    />
                  </View>
                </View>
              )
            })}
          </View>
        ) : (
          <View className="items-center flex-1 justify-center basis-4/5 shrink-0 ">
            <View className="flex-row justify-center">
              <Text
                className="text-2xl font-medium "
                sx={{
                  color: '$lightText'
                }}
              >
                You haven't created any task
              </Text>
            </View>
          </View>
        ))}
    </BaseLayout>
  )
}

export default ThisWeekTasksScreen
