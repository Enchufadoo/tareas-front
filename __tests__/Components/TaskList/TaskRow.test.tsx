import React from 'react'

import TaskRow from '@/Components/TasksList/TaskRow'
import { mockTask } from '../../__mocks__/Services/Tasks'
import { renderWithProviders } from '@/Testing/TestUtil'
import { fireEvent } from '@testing-library/react-native'
import { capitalize } from '@/Util/StringUtil'
import { ellapsedTime, hours } from '@/Util/DateUtil'
import moment from 'moment'
import MockDate from 'mockdate'

describe('TaskRow', () => {
  it('Swipe events have been triggered', () => {
    const onNavigateToTask = jest.fn()
    const task = mockTask()

    const { getByText } = renderWithProviders(
      <TaskRow index={1} onNavigateToTask={onNavigateToTask} item={task} />
    )

    const element = getByText(task.title)
    expect(element).toBeTruthy()

    fireEvent(element, 'swipeableWillOpen')
    fireEvent(element, 'swipeableWillClose')
  })

  it('On press will navigate to task', () => {
    const onNavigateToTask = jest.fn()
    const task = mockTask()

    const { getByText } = renderWithProviders(
      <TaskRow index={1} onNavigateToTask={onNavigateToTask} item={task} />
    )

    const element = getByText(task.title)
    expect(element).toBeTruthy()

    fireEvent.press(element)
    expect(onNavigateToTask).toBeCalled()
  })

  it('Description and date are rendered correctly', () => {
    const onNavigateToTask = jest.fn()
    const task = mockTask()

    const { getByText } = renderWithProviders(
      <TaskRow index={1} onNavigateToTask={onNavigateToTask} item={task} />
    )

    expect(getByText(task.description)).toBeTruthy()

    const lastProgress = capitalize(ellapsedTime(task.last_progress.created_at))

    expect(getByText(lastProgress)).toBeTruthy()
    expect(getByText(hours(moment(task.last_progress.created_at)))).toBeTruthy()
  })

  /**
   * Skipping this test due to this issue:
   * https://github.com/software-mansion/react-native-gesture-handler/issues/605
   * Need to create my own serializer to ignore the prop handlerTag that
   * gets added dynamically or there's something is wrong that I don't get
   * @todo
   */
  it.skip('renders correctly', () => {
    MockDate.set(new Date('2024-01-01T00:00:00Z'))

    const onNavigateToTask = jest.fn()
    const task = mockTask({
      id: 259,
      title: 'Despecto utpote bardus amo corrupti vis quisquam .',
      description: 'Tutis temperantia ver territo assentator admirati.',
      created_at: '2023-06-20T06:57:07.515Z',
      created_at_date: '2023-10-29',
      hours_since_progress: 20,
      progress_today: false,
      last_progress: { created_at: '2024-04-01T21:37:41.470Z', id: 296 }
    })

    const { toJSON } = renderWithProviders(
      <TaskRow index={1} onNavigateToTask={onNavigateToTask} item={task} />
    )

    const component = toJSON()

    expect(component).toMatchSnapshot()

    MockDate.reset()
  })
})
