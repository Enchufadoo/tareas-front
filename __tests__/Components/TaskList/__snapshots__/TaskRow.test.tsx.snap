// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskRow renders correctly 1`] = `
<View
  collapsable={false}
  handlerTag={4}
  handlerType="PanGestureHandler"
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  style={
    {
      "marginBottom": 10,
      "overflow": "hidden",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "bottom": 0,
        "flexDirection": "row",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": [
          {
            "translateX": -10000,
          },
        ],
      }
    }
  >
    <RNGestureHandlerButton
      activeOpacity={0.105}
      collapsable={false}
      delayLongPress={600}
      handlerTag={5}
      handlerType="NativeViewGestureHandler"
      onActiveStateChange={[Function]}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      onPress={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#5c5de5",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "maxWidth": 80,
          "width": "100%",
        }
      }
      underlayColor="black"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "black",
            "borderBottomLeftRadius": undefined,
            "borderBottomRightRadius": undefined,
            "borderRadius": undefined,
            "borderTopLeftRadius": undefined,
            "borderTopRightRadius": undefined,
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <View
        collapsable={false}
        style={
          [
            {
              "transform": [
                {
                  "scale": 0,
                },
              ],
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "white",
                "fontSize": 32,
              },
              undefined,
              {
                "fontFamily": "ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          î¬¢
        </Text>
      </View>
    </RNGestureHandlerButton>
    <View
      onLayout={[Function]}
    />
  </View>
  <View
    collapsable={false}
    handlerTag={6}
    handlerType="TapGestureHandler"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    pointerEvents="auto"
    style={
      {
        "transform": [
          {
            "translateX": -0,
          },
        ],
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "elevation": 1,
            "padding": 0,
            "paddingLeft": 0,
          },
        ]
      }
    >
      <View>
        <View>
          <View>
            <Text>
              Despecto utpote bardus amo corrupti vis quisquam .
            </Text>
            <Text>
              Tutis temperantia ver territo assentator admirati.
            </Text>
          </View>
          <View />
        </View>
      </View>
      <View>
        <View
          style={
            [
              {
                "borderBottomColor": "#AAA",
                "borderBottomWidth": 0.5,
              },
            ]
          }
        />
        <View>
          <Text>
            <Text>
              -92 days ago
            </Text>
            <Text>
               18:37
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
